apps/passport-client/components/screens/FrogScreens/Button.tsx

```diff
@@ -77,16 +83,56 @@ export function ActionButton({
   }, []);
 
   return (
-    <Button
+    <ButtonComponent
       onClick={handleClick}
       disabled={loading || disabled}
       pending={loading}
     >
       {children}
-    </Button>
+    </ButtonComponent>
   );
 }
 
+/**
+ * The FrogSearchButton allows a frog confetti animation when the button is disabled.
+ */
+export const FrogSearchButton = ({
+  disabled,
+  children,
+  ...props
+}: React.ComponentPropsWithRef<typeof Button>) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const container = useFrogParticles(ref);
+
+  useEffect(() => {
+    if (!container) {
+      return;
+    }
+
+    if (disabled) {
+      container.start();
+    }
+
+    return () => {
+      container.stop();
+    };
```

Comment:
Are there any issues if container.stop() is called without the container.start()? this would happen if `disabled = false`. Maybe to be safe / clear, we could just do this
```suggestion
    if (disabled) {
      container.start();

      return () => {
        container.stop();
      };
    }
```

Reply:
I don't think so but yes this looks better

Reply:
actually. we need to do like this so the particle effects are disabled first. added a comment